src/lib.rs

```diff
@@ -0,0 +1,59 @@
+mod cli;
+mod config;
+mod types;
+
+pub use crate::config::Config;
+use actix_web::{
+    error,
+    web::{self, Json},
+    App, Error, HttpServer,
+};
+use tempfile::TempDir;
+use tokio::{io::AsyncWriteExt, process::Command};
+use types::{SolToUmlRequest, SolToUmlResponse};
+
+async fn sol_to_uml(data: Json<SolToUmlRequest>) -> Result<Json<SolToUmlResponse>, Error> {
+    let data = data.into_inner();
+    let contract_dir = TempDir::new()?;
+    let contract_path = contract_dir.path();
+
+    for (name, content) in data.sources {
```

Comment:
actually you can make it even better
instead of `spawn`, use `spawn_blocking` (it still will return some kind of join handle), and then change all the `tokio::fs` operations to `std::fs`
because `tokio::fs` is basically calling `spawn_blocking` in each function, and it should be slower, than calling it once (per file)
so it should look something like
```rust
let join = data.sources.into_iter().map(|(name, content)| {
 tokio::task::spawn_blocking(async move {
    let file_path = contract_path.join(name);
    let prefix = file_path.parent();
    if let Some(prefix) = prefix {
        std::fs::create_dir_all(prefix)?;
    }
    let mut f = std::fs::File::create(file_path)?;
    f.write_all(content.as_bytes())?;
 })
});
futures::future::join_all(join);
```